crates/bevy_render/src/camera/camera.rs

```diff
@@ -22,38 +22,144 @@ use bevy_transform::components::GlobalTransform;
 use bevy_window::{WindowCreated, WindowId, WindowResized, Windows};
 use serde::{Deserialize, Serialize};
-use std::borrow::Cow;
+use std::{borrow::Cow, ops::Range};
 use wgpu::Extent3d;
 
+/// Render viewport configuration for the [`Camera`] component.
+///
+/// The viewport defines the area on the screen that the camera renders its image to.
+/// You can overlay multiple cameras in a single window using viewports to create effects like like split screen, minimaps, portals and character viewers.
+// TODO: remove reflect_value when possible
+#[derive(Reflect, Debug, Clone, Serialize, Deserialize)]
+#[reflect_value(Default, Serialize, Deserialize)]
+pub struct Viewport {
+    /// The physical position to render this viewport to within the [`RenderTarget`] of this [`Camera`].
+    /// (0,0) corresponds to the top-left corner
+    pub physical_position: UVec2,
+    /// The physical size of the viewport rectangle to render to within the [`RenderTarget`] of this [`Camera`].
+    /// The origin of the rectangle is in the top-left corner.
+    pub physical_size: UVec2,
+    /// The minimum and maximum depth to render to (on a scale from 0.0 to 1.0).
+    pub depth: Range<f32>,
+}
+
+impl Default for Viewport {
+    fn default() -> Self {
+        Self {
+            physical_position: Default::default(),
+            physical_size: Default::default(),
+            depth: 0.0..1.0,
+        }
+    }
+}
+
+/// Information about the current [`RenderTarget`].
+#[derive(Default, Debug, Clone)]
+pub struct RenderTargetInfo {
+    /// The physical size of this render target (ignores scale factor).
+    pub physical_size: UVec2,
+    /// The scale factor of this render target.
+    pub scale_factor: f64,
+}
+
+/// Holds internally computed [`Camera`] values.
+#[derive(Default, Debug, Clone)]
+pub struct ComputedCameraValues {
+    projection_matrix: Mat4,
+    target_info: Option<RenderTargetInfo>,
+}
+
 #[derive(Component, Debug, Reflect, Clone)]
 #[reflect(Component)]
 pub struct Camera {
-    pub projection_matrix: Mat4,
-    pub logical_target_size: Option<Vec2>,
-    pub physical_target_size: Option<UVec2>,
+    /// If set, this camera will render to the given [`Viewport`] rectangle within the configured [`RenderTarget`].
+    pub viewport: Option<Viewport>,
+    /// Cameras with a lower priority will be rendered before cameras with a higher priority.
     pub priority: isize,
+    /// If this is set to true, this camera will be rendered to its specified [`RenderTarget`]. If false, this
+    /// camera will not be rendered.
     pub is_active: bool,
+    /// The method used to calculate this camera's depth. This will be used for projections and visibility.
+    pub depth_calculation: DepthCalculation,
+    /// Computed values for this camera, such as the projection matrix and the render target size.
     #[reflect(ignore)]
-    pub target: RenderTarget,
+    pub computed: ComputedCameraValues,
+    /// The "target" that this camera will render to.
```

Comment:
```suggestion
    /// The "target" into which this camera will render.
```

Reply:
I think I prefer "to" over "into" here. I also like the flow of the original. "to which" feels a bit unnatural and formal to me.

Reply:
As always, english grammar is polarizing :)

Reply:
To be fair, I’m split in this internally. Ending sentences with prepositions is grammatically bad. But everyone commonly speaks and writes that way, which makes it more accessible and has a better tone. I always feel awkward saying ‘into which’ or whatever, and I’ve had to think ahead about the full sentence to know if it was necessary to be able to adjust and say it that way. Bleh. Happy to go along with honestly simpler language, even if my training also makes my brain poke me when I see prepositions at the end of sentences.